lua/telescope/actions/init.lua

```diff
@@ -257,6 +265,44 @@ actions.insert_symbol = function(prompt_bufnr)
   vim.api.nvim_put({selection.value[1]}, '', true, true)
 end
 
+actions.create_file = function(prompt_bufnr)
+  local file = vim.fn.input('Create file or directory(add ' ..
+                            path.separator ..
+                            ' at the end of file). Enter name: ')
+  if file == "" then return end
+
+  local fpath = actions.get_selected_entry(prompt_bufnr).path
+  if string.sub(fpath, -1) == path.separator then
+    fpath = vim.fn.fnamemodify(fpath, ':h:h')
+  else
+    fpath = vim.fn.fnamemodify(fpath, ':h')
+  end
+
+  fpath = fpath .. path.separator .. file
+
+  if string.sub(fpath, -1) ~= path.separator then
+    actions.close(prompt_bufnr)
+    path.create_file(fpath)
+    vim.cmd(string.format(':e %s', fpath))
```

Comment:
Missing ways to do split, etc.

Could do similar strategy we did for selecting files here. Maybe we need some sort of abstraction over that to make it easier generally.

Reply:
Hmmm i don't think so this is a function that only creates files. Not that anything is final. I was just playing around with it a bit more but how i thought this should work:

- keypress -> vim.fn.input filename
  - if filename == test/ ends with slash directory and navigate into that dir
  - if filename == test.lua create file and open it. We don't really have a way to specifiy the split/vsplit here
  - if filename == test/nested/ create both dirs
  - if filename == test/nested/test.lua create dirs and open file